services/order.py

```diff
@@ -0,0 +1,46 @@
+from django.db import transaction
+from db.models import Order, Ticket, User, MovieSession
+from django.utils import timezone
+from django.core.exceptions import ObjectDoesNotExist
+from django.db.models import QuerySet
+
+
+@transaction.atomic
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: str | None = None
+) -> Order | None:

```

Comment:
The return type hint 'Order | None' is used here, which is correct for Python 3.10 and above. Ensure that your environment supports this syntax. If not, consider using 'Optional[Order]' from the 'typing' module.